packages/rlc-common/src/buildClient.ts

```diff
@@ -378,6 +385,21 @@ export function getClientFactoryBody(
         }${loggerOptions}${customHeaderOptions}${credentialsOptions}
       }`;
 
+  let apiVersionStatement: string = "";
+  // Set the default api-version when we have a default AND its position is query
+  if (
+    model.apiVersionInfo?.definedPosition === "query" &&
+    !!model.apiVersionInfo?.defaultValue
+  ) {
+    apiVersionStatement = `
+    const apiVersion = options.apiVersion ?? "${model.apiVersionInfo?.defaultValue}";
+    delete options.apiVersion;
```

Comment:
I would prefer we don't mutate the options object to avoid side-effects. You can do something like this

```ts
   const {apiVersion, ...otherOptions}  = options;

  // replace the current use of options for `otherOptions` for example:

  const getClient = `const client = getClient(
        endpointUrl, ${credentialsOptions ? "credentials," : ""} otherOptions
      ) as ${clientTypeName};
```

Reply:
Let's try do this, but open to keep the delete if it turns out to be a big effort/risk

Reply:
I think another option is to spread at the createClient signature. For example

```ts
   export function createClient(endpoint: string, {apiVersion, ...options}?: ClientOptions)
```

If you do it this way you don't need to rename to `otherOptions` as the rest spread will contain everything but the properties before it.

Reply:
[Here is an example of why it is a good practice to avoid functions with side-effects ](https://www.typescriptlang.org/play/?#code/JYOwLgpgTgZghgYwgAgGIHt3IN4ChnLogQD8AXMgM5hSgDmANPsmAO7oXW0iPNgAWUCBE416TAjHQBXKKO51cAX1yqY0kAjDAiyKegAU6AA7ailChnQBKCgDd0wACY5mTiABsIkQqZ0hKADoiCGYEc3QvQI90OgMAAwART28UEzMA4OJkUEpnFDg9DS1-eOtlVVx1TQy9TAA5dABlfIBRGBgILQNsEIZkQMH0-0olS0xbZAdnVwJgGAMjP3NkOEpVkABPayyIa1mCFkF0VmRiU9aoKHQoAwAiYZXKfhkPJxAAcjBkfjg7FAEaWyxmuxmgYE2yAARl04NJKChgN9WGtkO5wgEaNItBAXKAjig8nQQHAwLIIHdygQVCpKhjKJEINFYvcALzsjmc5AAdQAkgAVAASVHyyAgHS6YHWnJlrMpqnp30eAWQrIOUlkFDuABY7hIjkIRMg7gJDXq+OwtQAmc0EEJawXQCAAQjuFVw9MZzLiyqCIXK+iWGUo5VwAHow8h+fwUO4vLVcqLAUUav4fqjCnl3GKJVpCNIpaKketKOEweHI76zlgYjxoOn1iFAshed9gOtCnRMC4QYhtEgWFg-o4XMUIJQEXAoJCsyhxZ0tJQPREojEfctMv6FSumWu2bKOTykS8CyLs-PJdKD+z5cvMb5g1bVeqZHJjbr9abhFqvxTP5bjRtfV7WNR0hFdd1PVXFlfStXYAwaZo2lzMAgxGK1QwjZBGjPOcUMofpYKoE83mhAp1lYWNxVAXEcjAO8GWg9dH3g3AgA)

Reply:
Thanks for the suggestion, I will change it this way.